qiita_db/handlers/prep_template.py

```diff
@@ -63,3 +66,16 @@ def get(self, prep_id):
             response = {'data': pt.to_dataframe().to_dict(orient='index')}
 
         self.write(response)
+
+
+class PrepTemplateAPItestHandler(OauthBaseHandler):
+    @authenticate_oauth
+    def post(self):
+        prep_info_dict = loads(self.get_argument('prep_info'))
```

Comment:
Fair point - I've investigated it with @ElDeveloper 
The name of the function (`get_argument`) seems a bit misleading here. The information is sent in the body of the request, so we are not limited at the argument of the URL. Here is the code snippet that we use to test this, sending a fake template of ~74Mb (which I don't know how often we see templates this big):
```python
import requests
from json import dumps
from qiita_client import QiitaClient

metadata_dict = {
    'SKB8.640193': {
        'primer': 'GTGCCAGCMGCCGCGGTAA',
        'barcode': 'GTCCGCAAGTTA',
        'platform': 'ILLUMINA',
        'instrument_model': 'Illumina MiSeq'},
    'SKD8.640184': {
        'primer': 'GTGCCAGCMGCCGCGGTAA',
        'barcode': 'GTCCGCAAGTTA',
        'platform': 'ILLUMINA',
        'instrument_model': 'Illumina MiSeq'}}
# Create a mapping file of ~76MB
metadata_dict['SKB8.640193']['primer'] = metadata_dict['SKB8.640193']['primer'] * 200000
metadata_dict['SKB8.640193']['primer'] = metadata_dict['SKB8.640193']['primer'] * 20
data = {
    'prep_info': dumps(metadata_dict),
    'study': 1,
    'data_type': '16S'}
CLIENT_ID = '19ndkO3oMKsoChjVVWluF7QkxHRfYhTKSFbAVt8IhK7gZgDaO4'
CLIENT_SECRET = ('J7FfQ7CQdOxuKhQAf1eoGgBAE81Ns8Gu3EKaWFm3IO2JKh'
                 'AmmCWZuabe0O5Mp28s1')
qclient = QiitaClient("https://localhost:21174", CLIENT_ID, CLIENT_SECRET,
                      server_cert='/Users/jose/qiime_software/QiiTa/qiita_core/support_files/server.crt')
r = qclient._request_oauth2(requests.post, 'https://localhost:21174/apitest/prep_template/', data=data, verify=False)
r.request.headers

Out[24]: {'Content-Length': '76000409', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'User-Agent': 'python-requests/2.8.1', 'Connection': 'keep-alive', 'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': u'Bearer ZVvdQbf0Xyp5WWzqH0XRuYikfktsVF5jJ2hJBfJNjoEwup4GVQJKfbG'}

r.request.url
Out[28]: 'https://localhost:21174/apitest/prep_template/'
```